### ISSUE TITLE
External Monitor

### ISSUE THREAD

**Issue by igormclven** [jsassu20/Lenovo-ThinkPad-T450-Mojave#7]
Hello, this time my problem is this:

I'm testing the latest build available, and I have problems with the headphones and DisplayPort output, I don't know if any additional adjustments are required. So far, I've replaced the EFI partition with this build and that's it.

It's been a while since I've used the headphones and there has always been this bug, it happens that when you connect these, you hear static, but when you rest the laptop, and returns from rest detects the headphones and work well, as long as you do not disconnect them, in that case, you would have to put the equipment to rest again. 

And not with respect to the output by DisplayPort, it happens that the screen becomes black, in the external monitor and in the laptop monitor, and then there is no way to return, you must force the restart.

I don't know if you have any clues to this. Thank you very much.

**Comment by shm007g** [comment]
As i know, t450 with i5 and i7 got vga and mini-dp port for external display.

Did this EFi work well on vga or mini-dp?

**Comment by jsassu20** [comment]
https://github.com/jsassu20/Lenovo-ThinkPad-T440-Mojave/tree/master/utilities/Audio%20Configurations/ALC3232/ALC%20Fix

Point your terminal to the folder I linked above after downloading the archive and run the install script inside then reboot and this will solve the audio plug.

As for the display out, I don't have a T440 and my graphics card is different from yours so I cant offer you a method to fix yours because I've never had a T440 for which I could spend the time needed to configure. With my T450 it was necessary that I maintain similar settings between the 2 sound controllers present in my system with respect to the layout-id of both needing to be identical. The 2 sound devices need to be names HDEF (main sound controller) HDAU (HDMI controller. Both SSDT files also need the presence of hda-gfx definitions which match each other.  The setting for both should be "onboard-1" (for the hda-gfx ACPI configuration.

You'll also need a properly setup IGPU configuration beyond just the normal options. There needs to be a correct configuration for framebuffer options based on the t440's Haswell based graphics and its use of AppleIntelFramebufferAZUL.kext instead of AppleIntelFramebufferBDW.kext which is what I had to configure all on my own  to get the settings working. 

Keep in mind as well that your graphics controller needs to be named IGPU (if its currently named "GFX0" or "VID"

The system audio controller needs to be named HDEF (if ts named AZAL or some other name in your configuration.

The HDMI device controller needs to be named HDAU (may be named B0D3)

Antd the system needs the IMEI device to exist.
